haskell-tests/Types/PayloadSerializationSpec.hs

```diff
@@ -62,7 +62,11 @@ checkPayload spv e =
 
 modifyPayloadBitmap :: (Word16 -> Word16) -> BS.ByteString -> BS.ByteString
 modifyPayloadBitmap f bs =
-    let Right ((header, bitmap), rest) = S.runGetState ((,) <$> S.getWord8 <*> S.getWord16be) bs 0
+    let res = S.runGetState ((,) <$> S.getWord8 <*> S.getWord16be) bs 0
+        ((header, bitmap), rest) = case res of
+            Right v -> v
+            -- This does not happen
```

Comment:
That is only true if the input bytestring has at least 24 bytes. State that as a precondition I suppose.